src/cart/crt_group.c

```diff
@@ -3019,23 +3030,20 @@ crt_group_secondary_create(crt_group_id_t grp_name, crt_group_t *primary_grp,
 		D_GOTO(out, rc);
 	}
 
-	found = false;
 	/* Record secondary group in the primary group */
-	for (i = 0; i < CRT_MAX_SEC_GRPS; i++) {
-		if (grp_priv_prim->gp_priv_sec[i] == NULL) {
-			found = true;
-			grp_priv_prim->gp_priv_sec[i] = grp_priv;
-			break;
-		}
-	}
-	if (!found) {
-		D_ERROR("Exceeded secondary groups limit\n");
-		D_GOTO(out, rc = -DER_NONEXIST);
+	D_ALLOC_PTR(entry);
+	if (entry == NULL) {
+		D_ERROR("Failed to allocate entry for group\n");
+		D_GOTO(out, rc = -DER_NOMEM);
 	}
+
+	entry->gps_priv = grp_priv;
+	d_list_add_tail(&entry->gps_link, &grp_priv_prim->gp_sec_list);
```

Comment:
Do we need locking for this list?

Reply:
yep we do. in fact we also have issue with locks - two different locks protecting same structure. changing in next patch to use same lock everywhere